### PR TITLE
Issue/71 export private key

### PR DESCRIPTION
**Pull request by willgarrett64**
For #71

### CODE REVIEW

File: src/components/HeaderTools.vue [hunk 1]
```diff
@@ -56,10 +56,11 @@
   import {SupportIcon} from "@heroicons/vue/solid"
   import vClickOutside from 'click-outside-vue3'
   import ForgetWallet from './index/ForgetModal.vue'
+  import ExportKey from './index/ExportModal.vue'
 
   export default {
     name: "HeaderTools",
-    components: {ArchiveIcon, ChevronDownIcon, CogIcon, KeyIcon, ForgetWallet, LockOpenIcon, LogoutIcon, SupportIcon},
+    components: {ArchiveIcon, ChevronDownIcon, CogIcon, KeyIcon, ForgetWallet, ExportKey, LockOpenIcon, LogoutIcon, SupportIcon},
```
Comment (annybs) [inline src/components/HeaderTools.vue:63]: I think this line is getting pretty long, so let's break it up and put each component on its own line - and then alphabetize 👍

File: src/components/index/ExportModal.vue [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import { LockOpenIcon, ShieldExclamationIcon, ClipboardCopyIcon } from '@heroicons/vue/outline'
+import { mapState } from 'vuex'
+
+export default {
+  name: 'ExportKey',
+  components: {
+    Modal,
+    LockOpenIcon,
+    ShieldExclamationIcon,
+    ClipboardCopyIcon
```
Comment (annybs) [inline src/components/index/ExportModal.vue:92]: Alphabetize

File: src/components/index/ExportModal.vue [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  },
+  props: {
+    close: Function,
+    visible: Boolean
+  },
+  computed: {
+    ...mapState({
+      address: 'address',
+    }),
```
Comment (annybs) [inline src/components/index/ExportModal.vue:114]: Change this to `...mapState(['address']),` as we don't need to explicitly alias the prop. See: https://vuex.vuejs.org/guide/state.html#the-mapstate-helper

File: src/components/index/ExportModal.vue [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    }
+  },
+  methods: {
+    cancel() {
+      this.reset()
+      this.close()
+    },
+    reset() {
+      this.password = ''
+      this.privateKey = ''
```
Comment (annybs) [inline src/components/index/ExportModal.vue:126]: Reset `passwordError` as well

File: src/components/index/ExportModal.vue [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      else {
+        this.passwordError = 'Incorrect password.'
+        return false
+      }
+    },
+    async exportKey() {
+      if (!await this.v$.$validate()) return
+      if (!await this.checkPassword()) return
+      
+      this.privateKey = await storage.getPrivateKey(this.password);
```
Comment (annybs) [inline src/components/index/ExportModal.vue:144]: Remove EOL semicolon, we don't use them in this project.

(I'm aware the existing code is a bit inconsistent about this, I'm counting on adding linting at some point to help with this. At the very least, that's the direction we're moving)

File: src/components/index/ExportModal.vue [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        </div>
+      </div>
+    </template>
+
+    <template v-slot:footer>
+      <div v-if="!privateKey" class="grid grid-cols-1 gap-24 px-24 pt-24 border-gray-700 border-solid md:grid-cols-2 border-t-default border-opacity-30 pb-24">
+        <button class="w-full button button--outline-success" @click="cancel">Cancel</button>
+        <button class="w-full button button--success" :disabled="!canSubmit" @click="exportKey">Export</button>
+      </div>
+      <div v-else class="px-24 pt-5 pb-30">
```
Comment (annybs) [inline src/components/index/ExportModal.vue:71]: The top border for the footer block is lost here. For continuity, preserve all CSS classes from the original `v-if` (with allowances for padding changes)

**Review by annybs — CHANGES_REQUESTED** [review]

File: src/components/index/ExportModal.vue (hunk 6 above)
Comment (willgarrett64) [reply src/components/index/ExportModal.vue]: I conserved all except md:grid-cols-2 to keep the button centralised. Not sure exactly what all these tailwind classes do, but it seemed to do the trick!

**Review by annybs — APPROVED** [review]
LGTM. Go ahead and merge, and it should automatically deploy to testnet. Ask @adamkdean to have a look once it's up in case he has any language or design feedback.

**Pull request merged**